File: M3Tools/Types/Events/ListenerEvents.vb
Namespace Events.Listeners
	Public Delegate Sub ListenerEventHandler(sender As Object, e As ListenerEventArgs)
	'Public Delegate Sub EditListenerEventHandler(sender As Object, e As ListenerEventArgs)

	Public Class ListenerEventArgs
		Inherits BaseArgs

		Property Listener As Types.Listener

		Sub New(listener As Types.Listener, Optional eventType As EventType = Nothing)
			Me.Listener = listener
			Me.EventType = eventType
		End Sub
	End Class
End Namespace

File: M3Tools/Types/Events/ListenerEvents.cs

namespace SPPBC.M3Tools.Events.Listeners
{
	/// <summary>
	/// 
	/// </summary>
	/// <param name="sender"></param>
	/// <param name="e"></param>
	public delegate void ListenerEventHandler(object sender, DataEventArgs<Types.Listener> e);
	//public delegate void ListenerEventHandler(object sender, ListenerEventArgs e);

	/// <summary>
	/// 
	/// </summary>
    public class ListenerEventArgs : DataEventArgs<Types.Listener>
    {

		/// <summary>
		/// <inheritdoc/>
		/// </summary>
        public override Types.Listener Value { get; protected set; }

		/// <summary>
		/// <inheritdoc/>
		/// </summary>
		/// <param name="listener"></param>
		/// <param name="eventType"></param>
        public ListenerEventArgs(Types.Listener listener, EventType eventType = default) : base(listener, eventType)
        {
        }
    }
}
